### PR TITLE
Aero shield

### PR DESCRIPTION
**Pull request by cenculis**

### CODE REVIEW

**Comment by gergelytakacs** [comment]
Im back bit***s. @AnnaVargova  please do a proper code review before merging. Also, please make sure that the stuck Spellcheck CI run is mended, someone should have a look at that.

**Comment by gergelytakacs** [comment]
@cenculis Good job, sorry for taking one month with this. Guys and girls (@AnnaVargova ), if you wish to pretend agile SW development  Im up for it. But in that case issues etc. must be used and i am actually happy to communicate. It can possibly go easier than on Teams.

**Comment by gergelytakacs** [comment]
@AnnaVargova Has a code review been performed for this PR?

**Comment by gergelytakacs** [comment]
@cenculis please make your Thesis supervisor do a pull review.

**Comment by gergelytakacs** [comment]
@cenculis 

- Matlab functions bear the same names as the filenames
- Im pretty sure there is a mapping function doing the same exact thing somewhere already
- Even if not, it is a good idea to add these functions somewhere in the common area of the code repo so that others can use it
image

File: src/AeroShield.cpp
Comment (gergelytakacs) [inline src/AeroShield.cpp:51]: Variables must be written in camelCase (https://en.wikipedia.org/wiki/Camel_case)

File: matlab/examples/AeroShield/AeroShield_open_loop/AeroShield_open_loop.m
Comment (gergelytakacs) [inline matlab/examples/AeroShield/AeroShield_open_loop/AeroShield_open_loop.m:3]: is clear a necessary after clear all?

File: examples/AeroShield/AeroShield_OpenLoop/AeroShield_OpenLoop.ino
Comment (gergelytakacs) [inline examples/AeroShield/AeroShield_OpenLoop/AeroShield_OpenLoop.ino:13]: CamelCase

**Review by gergelytakacs — CHANGES_REQUESTED** [review]
Please address changes here. Also this PR is outdated please merge your work more frequently

**Review by cenculis — COMMENTED** [review]
Thanks a lot for checking the code.

**Comment by gergelytakacs** [comment]
> Thanks a lot for checking the code.

You are very welcome. As I said , the more you  will use Git and GitHub issues, the more I can help.

Now the thing is, you are developing software for hardware, so please keep in mind, that I cant really test your code - i can just do a visual inspection. Code review would mean that someone actually tests your stuff, but its better than nothing.

File: examples/AeroShield/AeroShield_PID/AeroShield_PID.ino
Comment (gergelytakacs) [inline examples/AeroShield/AeroShield_PID/AeroShield_PID.ino:49]: Same as in the open loop example. Im thinking this should be just begin() and the rest is done automatically

File: examples/AeroShield/AeroShield_PID/AeroShield_PID.ino
Comment (gergelytakacs) [inline examples/AeroShield/AeroShield_PID/AeroShield_PID.ino:50]: CamelCase

File: examples/AeroShield/AeroShield_PID/AeroShield_PID.ino
Comment (gergelytakacs) [inline examples/AeroShield/AeroShield_PID/AeroShield_PID.ino:51]: CamelCase

File: matlab/AeroShield.m
Comment (gergelytakacs) [inline matlab/AeroShield.m:2]: Say what your code does. :)

File: examples/AeroShield/AeroShield_OpenLoop/AeroShield_OpenLoop.ino
Comment (gergelytakacs) [inline examples/AeroShield/AeroShield_OpenLoop/AeroShield_OpenLoop.ino:14]: image

File: matlab/examples/AeroShield/AeroShield_PID/AeroShield_PID.m
Comment (gergelytakacs) [inline matlab/examples/AeroShield/AeroShield_PID/AeroShield_PID.m:13]: Camel Case !

File: matlab/examples/AeroShield/AeroShield_PID/AeroShield_PID.m
Comment (gergelytakacs) [inline matlab/examples/AeroShield/AeroShield_PID/AeroShield_PID.m:20]: That is some very careful tuning:) Its okay, im just in awe, how exact you have gone. This raises a question: Imagine you create 100 Aero Shield devices. Would this tunign be good for all of them?:)

File: matlab/examples/AeroShield/AeroShield_speedTest/AeroShield_speedTest.m
Comment (gergelytakacs) [inline matlab/examples/AeroShield/AeroShield_speedTest/AeroShield_speedTest.m:5]: camel case starts with small letter

Comment (gergelytakacs) [reply matlab/examples/AeroShield/AeroShield_speedTest/AeroShield_speedTest.m:5]: I mean lower case.

File: src/AeroShield.cpp
Comment (gergelytakacs) [inline src/AeroShield.cpp:3]: Tell what it does

File: src/AeroShield.cpp
Comment (gergelytakacs) [inline src/AeroShield.cpp:13]: State your name:)

File: src/AeroShield.h
Comment (gergelytakacs) [inline src/AeroShield.h:3]: Say what it does

File: src/AeroShield.h
Comment (gergelytakacs) [inline src/AeroShield.h:13]: Say my name, say my... (I mean yours.)
image

File: src/AeroShield.h
Comment (gergelytakacs) [inline src/AeroShield.h:15]: Say what it does, etc. like in the other files.

**Review by gergelytakacs — CHANGES_REQUESTED** [review]
Nice job. Please read the comments and correct minor things. Continue working and try to communicate via GitHub issies etc., so I can help more.

**Comment by gergelytakacs** [comment]
@cenculis Make your thesis supervisor make you use GitHub Issues and agile more;)

**Comment by gergelytakacs** [comment]
@AnnaVargova 
image

image

File: examples/AeroShield/AeroShield_OpenLoop/AeroShield_OpenLoop.ino
Comment (cenculis) [reply examples/AeroShield/AeroShield_OpenLoop/AeroShield_OpenLoop.ino:13]: I think there is some kind of a problem with me uploading the stuff on GIT. I changed the camelCase variables before you corrected me! but the shitty thing is still there... Im gonna ask anka about it because I really dont know where the problem is. Also the same is about spelling milliseconds :D

File: examples/AeroShield/AeroShield_OpenLoop/AeroShield_OpenLoop.ino
Comment (cenculis) [reply examples/AeroShield/AeroShield_OpenLoop/AeroShield_OpenLoop.ino:22]: I think it can be in the begin methode, this is just the way I wrote it the first time :D ... gonna try

File: examples/AeroShield/AeroShield_OpenLoop/AeroShield_OpenLoop.ino
Comment (cenculis) [reply examples/AeroShield/AeroShield_OpenLoop/AeroShield_OpenLoop.ino:35]: Yes it does. If I dont need that Im gonna just get rid of it. It was just for my pleasure

File: examples/AeroShield/AeroShield_PID/AeroShield_PID.ino
Comment (cenculis) [reply examples/AeroShield/AeroShield_PID/AeroShield_PID.ino:12]: The Information Headers in code examples etc are there just as a reminder for me to write em... most of them right now are stolen from AutomationShield examples. I will try my best to correct them till the next pull request

File: examples/AeroShield/AeroShield_OpenLoop/AeroShield_OpenLoop.ino
Comment (cenculis) [reply examples/AeroShield/AeroShield_OpenLoop/AeroShield_OpenLoop.ino:14]: Im so so sorry .... :DDD really, its GitHubs fault

File: matlab/examples/AeroShield/AeroShield_PID/AeroShield_PID.m
Comment (cenculis) [reply matlab/examples/AeroShield/AeroShield_PID/AeroShield_PID.m:20]: oh hell no.... it wont work.... I was tunning that shit for 3 days and just now, retune it.... It works better now, with a bit more normal-ish numbers..

File: src/AeroShield.h
Comment (cenculis) [reply src/AeroShield.h:15]: the last meme is super cool :DD I wasnt communicating for the last 2 weeks.... so tomorrow I have a meeting with anka :) We gonna figure it out

File: examples/AeroShield/AeroShield_OpenLoop/AeroShield_OpenLoop.ino
Comment (gergelytakacs) [reply examples/AeroShield/AeroShield_OpenLoop/AeroShield_OpenLoop.ino]: All right. I think this is the part where you find out what went wrong and learn Git in the process:)

File: src/AeroShield.h
Comment (gergelytakacs) [reply src/AeroShield.h:16]: All right, well has that happened?:)

**Review by gergelytakacs — APPROVED** [review]

**Comment by gergelytakacs** [comment]
@cenculis  Merged! Woohoo! Do you want to continue working on this branch or check out a feature-specific one next time? (I can leave this for now, but Git use prefers creating branches for smaller, usage and function specific branches). Eg. say you want to do a PID control example in matlab - thats one branch, then review, merge, delete branch and go for next task

**Pull request merged**
